### PR TITLE
[JBPM-9700] Signal name not resolved when adding listener

### PR DESCRIPTION
**Pull request by fjtirado**
**JIRA**: 

[link](https://issues.redhat.com/browse/JBPM-9700)

### CODE REVIEW

**Review by gmunozfe — COMMENTED** [review]
Looks good to me, excellent work @fjtirado !

I also have a comment for the code below (not modified by this PR) that it's adding the _eventListener_ twice in case of variable expression. It doesn't seem correct at all (I think it's more clear as it's done in this PR with an "else clause):
https://github.com/kiegroup/jbpm/blob/62f92c98f5d3f68f08fd519638259be97aa2d574/jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java#L573-L576

File: jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java [hunk 1]
```diff
@@ -566,7 +567,12 @@ public TimerInstance configureSLATimer(String slaDueDateExpression, String timer
     protected void registerExternalEventNodeListeners() {
         for (Node node : getWorkflowProcess().getNodes()) {
             if (node instanceof EventNode && "external".equals(((EventNode) node).getScope())) {
-                addEventListener(((EventNode) node).getType(), EMPTY_EVENT_LISTENER, true);
+                String eventType = ((EventNode) node).getType();
+                if (isVariableExpression(eventType)) {
+                    addEventListener(resolveVariable(eventType), EMPTY_EVENT_LISTENER, true);
```
Comment (elguardian) [inline jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java:572]: if you modify the add... modify the remove unregisterExternalEventNodeListeners

**Comment by elguardian** [comment]
add test case (you have the bpmn2)
tbh honest I don't consider this a valid use case as we are tying this to a process variable which may vary dynamically to a external source.

**Review by gmunozfe — APPROVED** [review]

**Comment by elguardian** [comment]
probably will superseeded by
https://github.com/kiegroup/jbpm/pull/1909

**Pull request merged**